### DISCUSSION TITLE
Why does ILogger.BeginScope<TState>(TState) return IDisposable?

### DISCUSSION THREAD

**Discussion by paulomorgado** [dotnet/runtime#96125] [Q&A]
In what situations would `ILogger.BeginScope<TState>(TState)` return a `null` `IDisposable`?

**Comment by huoyaoyuan** [comment]
There are implementations that returns null. If you search at source.dot.net, you can find that `Microsoft.AspNetCore.Grpc.JsonTranscoding.IntegrationTests.Infrastructure.ForwardingLogger` returns null.

**Reply by paulomorgado** [reply]
That is dangerous.

It should return, at least, something like:

```csharp
private sealed class EmptyDisposable : IDisposable
{
    public static EmptyDisposable Instance { get; } = new EmptyDisposable();

    private EmptyDisposable()
    {
    }

    void IDisposable.Dispose()
    {
        // Interface implementation
    }
}
```

**Reply by colejohnson66** [reply]
Why is it dangerous? You should never be manually calling Dispose if you don’t need to. A `using` block will handle nulls for you.

**Comment by paulomorgado** [comment]
In hindsight, I might have overthought this.
